lsp/server/src/test/TestHelper.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { Stats } from 'fs';
+import { mkdtemp, rm, stat } from 'fs/promises';
+import * as os from 'os';
+import * as path from 'path';
+import * as process from 'process';
+import * as sinon from 'sinon';
+import Account from '../../testFixture/objectInfos/Account.json';
+import {
+    ConfigAggregator,
+    StateAggregator,
+    Connection,
+    AuthInfo
+} from '@salesforce/core';
+import { WorkspaceUtils } from '../utils/workspaceUtils';
+
+const testOrgConfig = {
+    key: 'target-org',
+    value: 'testOrg'
+};
+
+const noOrgConfig = {
+    key: 'target-org'
+};
+
+export class TempProjectDirManager {
+    readonly projectDir: string;
+
+    private constructor(projectDir: string) {
+        this.projectDir = projectDir;
+    }
+
+    async removeDir(): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            let projectDirStats: Stats;
+            try {
+                projectDirStats = await stat(this.projectDir);
+            } catch (err) {
+                return reject(
+                    `Project dir '${this.projectDir}' does not exist or is inaccessible.`
+                );
+            }
+            if (!projectDirStats.isDirectory()) {
+                return reject(
+                    `Project dir '${this.projectDir}' is not a directory.`
+                );
+            }
+            await rm(this.projectDir, { recursive: true, force: true });
+            return resolve();
+        });
+    }
+
+    static async createTempProjectDir(): Promise<TempProjectDirManager> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                const projectDir = await mkdtemp(
+                    path.join(os.tmpdir(), 'offlineWizard-')
+                );
+                return resolve(new TempProjectDirManager(projectDir));
+            } catch (err) {
+                return reject(err);
+            }
+        });
+    }
+}
```

Comment:
`TempProjectDirManager` used to be in root project

Reply:
This class should not be in the test helper. It's being accessed in the production code. So it is a first class citizen in the codebase.

What's yucky is the way it is setup you need to reexport this class from root's index.ts. Let's move it to a place because it deserves it. Then we can get rid of reexport.